src/PAModel/Serializers/SourceSerializer.cs

```diff
@@ -370,6 +370,11 @@ private static void LoadSourceFiles(CanvasDocument app, DirectoryReader director
                 // Json peer to a .pa file. 
                 var controlExtraData = file.ToObject<Dictionary<string, ControlState>>();
                 var topParentName = file._relativeName.Replace(".editorstate.json", "");
+
+                // Check for the designated parent control. Fall back on the file name.
+                if (controlExtraData.ContainsKey(TopParentObjectKey) && (controlExtraData[TopParentObjectKey].IsTopParent ?? false))
```

Comment:
So, why do we need this TopParentObjectKey? 
We could just serialize as is, and then the deserialize retrieval could be 
`controlExtraData.FirstOrDefault(kvp => kvp.Value.IsTopParent)) ?? topParentName`

Reply:
Restructured this to use a specific object model for editor state files.